x-pack/plugins/security_solution/server/lib/detection_engine/fleet_integrations/api/get_installed_integrations/installed_integration_set.ts

```diff
@@ -33,10 +33,59 @@ interface PackageInfo extends InstalledPackageBasicInfo {
 export const createInstalledIntegrationSet = (): IInstalledIntegrationSet => {
   const packageMap: PackageMap = new Map<string, PackageInfo>([]);
 
+  const addPackage = (fleetPackage: PackageListItem): void => {
+    if (fleetPackage.type !== 'integration') {
+      return;
+    }
+    if (fleetPackage.status !== 'installed') {
+      return;
+    }
```

Comment:
Now that we know ~ALL THE THINGS!!~ all the packages, we can probably return them (and their integrations) as well to fix another bug we have with showing an integration title when the integration is not installed. I.e. we show `O365` and not `Microsoft 365`. This should be a separate fix, of course, and I can open a ticket for that.

Reply:
Here we go! https://github.com/elastic/kibana/issues/152408

This fix will be interesting assuming a large number of uninstalled integrations. Probably not reasonable to return titles for all integrations(?), so may need to pass the rule's integrations to our API and only return those? Alternatively, perhaps we just need to add them to the prebuilt rules? 🤔